### PR TITLE
Add extra meta to wallet and ln operations

### LINKED ISSUE #3042: Add `extra_meta` to `LightningMeta` / `WalletOperationMeta`
`MintMeta` has a hook for external developers to add arbitrary metadata to a mint state machine: https://github.com/fedimint/fedimint/blob/109c7c1207b7c16d6d70bc7ac39cb178032471a8/modules/fedimint-mint-client/src/lib.rs#L511-L516

Would be nice to either add the same hook to `LightningMeta` and `WalletOperationMeta`, or just expect external developers to store metadata themselves.

### PR DESCRIPTION
**Pull request by elsirion**
Fixes #3042

### CODE REVIEW

File: gateway/ln-gateway/tests/integration_tests.rs [hunk 1]
```diff
@@ -1200,7 +1200,7 @@ async fn test_gateway_executes_swaps_between_connected_federations() -> anyhow::
                 "Gateway balance after swap".to_string(),
                 || async {
                     let post_balances = get_balances(&rpc, &[id1, id2]).await;
-                    if post_balances[0] == pre_balances[0] && post_balances[1] == pre_balances[1] {
+                    if post_balances[0] == pre_balances[0] || post_balances[1] == pre_balances[1] {
```
Comment (Maan2003) [inline gateway/ln-gateway/tests/integration_tests.rs:1203]: I don't understand

**Review by Maan2003 — APPROVED** [review]
but approval for rest of the PR.

File: gateway/ln-gateway/tests/integration_tests.rs (hunk 1 above)
Comment (elsirion) [reply gateway/ln-gateway/tests/integration_tests.rs:1203]: I should have linked this comment https://github.com/fedimint/fedimint/issues/3672#issuecomment-1824530248

Comment (Maan2003) [reply gateway/ln-gateway/tests/integration_tests.rs:1203]: oh this is not the exit condition of retry but retry again condition.
makes a lot more sense now.

**Review by Maan2003 — APPROVED** [review]
now for complete PR

**Comment by okjodom** [comment]
sorry, removed from queue then readded :p. I saw [this change](https://github.com/fedimint/fedimint/pull/3703#discussion_r1403577333) in gh mobile and needed more context on it, now caught up on the reasoning

**Review by okjodom — APPROVED** [review]

**Pull request merged**